tests/e2e/matmul/generate_code_mx.py

```diff
@@ -0,0 +1,341 @@
+
+from typing import Optional
+import argparse
+import dataclasses
+import typing
+
+from tests.e2e.matmul.common import *
+from tests.e2e.matmul.compilation_info import *
+
+
+# Helper for generate_function.
+# Generates a name for a test function in the generated MLIR code.
+def generate_function_name_mx(
+    lhs_rhs_type: MatrixElemTypeId,
+    acc_type: MatrixElemTypeId,
+    mx_scale_type: MatrixElemTypeId,
+    mx_block_size: int,
+    shapes: TestInputMatricesShapes,
+    accumulate: bool,
+    compilation_info: typing.Optional[CompilationInfo] = None,
+):
+    input_t = lhs_rhs_type.value
+    acc_t = acc_type.value
+    lhs_r = int_or_DYN(shapes.lhs_rows)
+    lhs_c = int_or_DYN(shapes.lhs_cols)
+    rhs_r = int_or_DYN(shapes.rhs_rows)
+    rhs_c = int_or_DYN(shapes.rhs_cols)
+    acc_r = int_or_DYN(shapes.acc_rows)
+    acc_c = int_or_DYN(shapes.acc_cols)
+
+    info = ""
+    if compilation_info:
+        info = f"_for_{compilation_info.dispatch_lowering_pass_pipeline}"
+
+    matmul_kind = "matmul_accumulate" if accumulate else "matmul"
+
+    return (
+        f"{matmul_kind}_mx_scale_{mx_scale_type}_{lhs_r}x{lhs_c // mx_block_size}x{mx_block_size}x{input_t}_times_"
+        + f"{rhs_r}x{rhs_c // mx_block_size}x{mx_block_size}x{input_t}_into_{acc_r}x{acc_c}x{acc_t}"
+    )
+
+
+# Generates a test function in the generated MLIR code.
+def generate_function_mx(
+    lhs_rhs_type: MatrixElemTypeId,
+    acc_type: MatrixElemTypeId,
+    mx_scale_type: MatrixElemTypeId,
+    mx_block_size: int,
+    shape: TestShape,
+    transpose_rhs: bool,
+    dynamicity: Dynamicity,
+    compilation_info: Optional[CompilationInfo] = None,
+):
+    if not transpose_rhs:
+        raise ValueError("MX tests require transpose_rhs")
+    if dynamicity != Dynamicity.STATIC:
+        raise ValueError("MX tests require static shape")
+
+    shapes = generate_shapes(shape, transpose_rhs, dynamicity)
+    func_name = generate_function_name_mx(
+        lhs_rhs_type=lhs_rhs_type,
+        acc_type=acc_type,
+        mx_scale_type=mx_scale_type,
+        mx_block_size=mx_block_size,
+        shapes=shapes,
+        accumulate=shape.accumulate,
+        compilation_info=compilation_info,
+    )
+    lhs_r = int_or_question_mark(shapes.lhs_rows)
+    lhs_c = int_or_question_mark(shapes.lhs_cols)
+    rhs_r = int_or_question_mark(shapes.rhs_rows)
+    rhs_c = int_or_question_mark(shapes.rhs_cols)
+    acc_r = int_or_question_mark(shapes.acc_rows)
+    acc_c = int_or_question_mark(shapes.acc_cols)
+
+    lhs_tensor_type = f"tensor<{lhs_r}x{lhs_c}x{lhs_rhs_type.value}>"
+    rhs_tensor_type = f"tensor<{rhs_r}x{rhs_c}x{lhs_rhs_type.value}>"
+    acc_tensor_type = f"tensor<{acc_r}x{acc_c}x{acc_type.value}>"
+
+    (
+        compilation_info_string,
+        compilation_info_attr,
+    ) = generate_compilation_info_string_and_attr(compilation_info)
+
+    lhs_scales_shape = [shape.m, shape.k // mx_block_size]
```

Comment:
We might want to test that `shape.k` is a multiple of 32?

(Unless we decide that `shape.k` doesn't have the `32` on it and that `mx_block_size` is a separate parameter ... which I think we should consider)

Reply:
Assertion added : https://github.com/iree-org/iree/pull/22170/commits/c0ae44a8056abd5304014bd954cfa005c9fbaee3